pkg/arvo/app/chat-hook.hoon

```diff
@@ -0,0 +1,353 @@
+::  chat-hook:
+::  mirror chat data from foreign to local based on read permissions
+::  allow sending chat messages to foreign paths based on write perms
+::
+/-  *permission-store, *chat-hook
+/+  *chat-json
+|%
++$  move  [bone card]
+::
++$  card
+  $%  [%diff [%chat-update chat-update]]
+      [%quit ~]
+      [%poke wire dock poke]
+      [%pull wire dock ~]
+      [%peer wire dock path]
+  ==
+::
++$  state
+  $%  [%0 state-zero]
+  ==
+::
++$  state-zero
+  $:  synced=(map path ship)
+      boned=(map wire (list bone))
+  ==
+::
++$  poke
+  $%  [%chat-action chat-action]
+      [%permission-action permission-action]
+  ==
+::
+--
+::
+|_  [bol=bowl:gall state]
+::
+++  this  .
+::
+++  prep
+  |=  old=(unit state)
+  ^-  (quip move _this)
+  ?~  old
+    :_  this
+    [ost.bol %peer /permissions [our.bol %permission-store] /updates]~
+  [~ this(+<+ u.old)]
+::
+++  poke-json
+  |=  jon=json
+  ^-  (quip move _this)
+  (poke-chat-action (json-to-action jon))
+::
+++  poke-chat-action
+  |=  act=chat-action
+  ^-  (quip move _this)
+  ?>  ?=(%message -.act)
+  ::  local
+  :_  this
+  ?:  (team:title our.bol src.bol)
+    ?.  (~(has by synced) path.act)
+      ~
+    =/  ship  (~(got by synced) path.act)
+    =/  appl  ?:(=(ship our.bol) %chat-store %chat-hook)
+    [ost.bol %poke / [ship appl] [%chat-action act]]~
+  ::  foreign
+  =/  ship  (~(get by synced) path.act)
+  ?~  ship
+    ~
+  ?.  =(u.ship our.bol)
+    ~
+  ::  scry permissions to check if write is permitted
+  ?.  (permitted-scry [(scot %p src.bol) %chat (weld path.act /write)])
+    ~
+  =.  author.envelope.act  src.bol
+  =.  when.envelope.act  now.bol
```

Comment:
Stylistically, I'd recommend:

```
=:  author.envelope.act  src.bol
    when.envelope.act  now.bol
  ==
```

Reply:
Thanks, made the change in all new files.